os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -555,8 +555,10 @@ void binary_manager_release_binary_sem(int bin_idx)
 		{
 			if (holder && holder->htcb && holder->htcb->group && holder->htcb->group->tg_binidx == bin_idx) {
 				/* Increase semcount and release itself from holder */
+				holder->counts = 1;
 				sem->semcount++;
 				sem_releaseholder(sem, holder->htcb);
```

Comment:
It is duplicated because sem_releaseholder finds tcb of holders by searching a holder list and then free holder.
So how about calling sem_freeholder(need to change it to public function) without line558?